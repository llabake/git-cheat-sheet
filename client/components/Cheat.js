import React, { Component } from 'react';
import { CopyToClipboard } from 'react-copy-to-clipboard';

export default class Cheat extends Component {
  state = {
    copied: false,
  };

  clicked = () => {
    this.setState({ copied: true }, () => {
      setTimeout(() => this.setState({ copied: false }), 1000);
    });
  };

  render() {
    const { cheat: { description, commands } } = this.props;
    const { copied } = this.state;
    const [keyword, ...body] = commands.split(' ');
    const remainder = body.join(' ');
    return (
      <div className="header">
        <span>{description}</span>
        <span style={{ display: `${copied ? 'inline' : 'none'}` }} className="cheat-copied">
          Copied!
        </span>
        <pre>
          $&nbsp;
          <code>
            <CopyToClipboard text={commands} onCopy={this.clicked}>
              <span>
                <span className="first">{keyword}</span>
                <span>
                &nbsp;
                  {remainder}
                </span>
              </span>
            </CopyToClipboard>
          </code>
        </pre>
      </div>
    );
  }
}
