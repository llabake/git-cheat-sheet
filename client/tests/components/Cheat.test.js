import React from 'react';
import { shallow, mount } from 'enzyme';
import Cheat from '../../components/Cheat';

const setup = () => {
  const props = {
    cheat: {
      description: 'description',
      commands: "git lorem ipsum",
    },
    clicked: jest.fn(() => new Promise(resolve => resolve())),
  };
  
  const wrapper = shallow(<Cheat {...props} />);
  
  return { wrapper, props };
};
describe('Cheat component', () => {
  const { wrapper } = setup();
  it('renders without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('copies a git command to clipboard', () => {
    const spy = jest.spyOn(wrapper.instance(), 'clicked');
    wrapper.instance().clicked();
    expect(spy).toHaveBeenCalled();
  });
});
