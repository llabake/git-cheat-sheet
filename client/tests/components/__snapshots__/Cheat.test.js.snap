// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cheat component renders without crashing 1`] = `
<div
  className="header"
>
  <span>
    description
  </span>
  <span
    className="cheat-copied"
    style={
      Object {
        "display": "none",
      }
    }
  >
    Copied!
  </span>
  <pre>
    $ 
    <code>
      <CopyToClipboard
        onCopy={[Function]}
        text="git lorem ipsum"
      >
        <span>
          <span
            className="first"
          >
            git
          </span>
          <span>
             
            lorem ipsum
          </span>
        </span>
      </CopyToClipboard>
    </code>
  </pre>
</div>
`;
